### PR TITLE
feat: action menu component

### PR DESCRIPTION
**Pull request by ebgranger**
## What I did

This is the new Action Menu Component.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -101,7 +101,6 @@
   ],
   "dependencies": {
     "@babel/standalone": "^7.17.11",
-    "@washingtonpost/wpds-assets": "^1.18.0",
```
Comment (artmsilva) [inline package.json:104]: 👀 What's this change for? I support it. I am curious

File: ui/action-menu/src/ActionMenuItem.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  display: "flex",
+  flexBasis: "auto",
+  flexDirection: "row",
+  justifyContent: "flex-start",
+  flexWrap: "none",
+  transition: `background ${theme.transitions.fast} ${theme.transitions.inOut}`,
+  width: "100%",
+  "&:hover": {
+    backgroundColor: theme.colors.alpha25,
+    cursor: "pointer",
```
Comment (wp-aberg) [inline ui/action-menu/src/ActionMenuItem.tsx:23]: image

I'm fairly certain we don't want the focus ring to be visible when we hover

```suggestion
     "&:focus-visible": {
      outline: "none",
     },
```

File: ui/action-menu/src/ActionMenuContent.tsx [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+  color: theme.colors.primary,
+  fontFamily: theme.fonts.subhead,
+  fontSize: theme.fontSizes["050"],
+  fontWeight: theme.fontWeights.regular,
+  lineHeight: theme.sizes["100"],
+  marginTop: theme.sizes["025"],
+  maxHeight: "var(--radix-dropdown-menu-content-available-height)",
+  maxWidth: "var(--radix-dropdown-menu-content-available-width)",
+  minWidth: "200px",
+  overflow: "auto",
```
Comment (wp-aberg) [inline ui/action-menu/src/ActionMenuContent.tsx:86]: adding 1px of padding here will allow the blue focus outline of the items not to be cut off

File: ui/action-menu/src/ActionMenuContent.tsx [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+
+import { ActionMenuPortal } from "./ActionMenuPortal";
+import { DensityProp } from "./ActionMenuRoot";
+
+const NAME = "ActionMenuContent";
+
+export const ContentDensityVariants = {
+  density: {
+    loose: {
+      "& .action-menu-item": {
```
Comment (wp-aberg) [inline ui/action-menu/src/ActionMenuContent.tsx:18]: I'm not sure about the use of descendent selectors here. I'd prefer to see these styles in the components they are affecting as opposed to the parent.

File: ui/action-menu/src/ActionMenuItem.tsx [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+import * as ActionMenuPrimitive from "@radix-ui/react-dropdown-menu";
+
+import { DropdownMenuItemProps as RadixDropdownMenuItemProps } from "@radix-ui/react-dropdown-menu";
+
+const NAME = "ActionMenuItem";
+
+export const ItemStyles = {
+  alignItems: "center",
+  background: theme.colors.secondary,
```
Comment (wp-aberg) [inline ui/action-menu/src/ActionMenuItem.tsx:13]: for when we have tabbed in and do want to see the focus outline, altering the corners to match the parent prevents that clipping

image

```
"&:first-child": {
    borderTopLeftRadius: `calc(${theme.radii["050"]} * 0.75)`,
    borderTopRightRadius: `calc(${theme.radii["050"]} * 0.75)`,
  },
  "&:last-child": {
    borderBottomLeftRadius: `calc(${theme.radii["050"]} * 0.75)`,
    borderBottomRightRadius: `calc(${theme.radii["050"]} * 0.75)`,
  },
  ```

**Comment by wp-aberg** [comment]
image

image

Another option with the focus ring is to favor the first screen shot style everywhere and not ever let the ring hit the edge the way it does on the second level menu

File: build.washingtonpost.com/docs/components/action-menu.mdx [hunk 6]
```diff
@@ -0,0 +1,241 @@
...
+### Portal
+
+Description of portal
+
+### Sub Menu
+
+Description of sub menu
+
+```jsx withPreview
+<p>test</p>
```
Comment (wp-aberg) [inline build.washingtonpost.com/docs/components/action-menu.mdx:168]: If these examples with `test` won't be completed before the first merge we should remove them. On merges to main the docs site is deployed immediately.

File: ui/action-menu/src/ActionMenuCheckboxItem.tsx [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
+import * as ActionMenuPrimitive from "@radix-ui/react-dropdown-menu";
+
+import { DropdownMenuCheckboxItemProps as RadixDropdownMenuCheckboxItemProps } from "@radix-ui/react-dropdown-menu";
+
+import { ItemStyles } from "./ActionMenuItem";
+
+const NAME = "ActionMenuCheckboxItem";
+
+export const StyledCheckboxItem = styled(ActionMenuPrimitive.CheckboxItem, {
```
Comment (wp-aberg) [inline ui/action-menu/src/ActionMenuCheckboxItem.tsx:13]: Exporting the styled primitive components like this in combination with the star export in the index means a bunch of pops tables get generated in the docs page. What's the intent of exporting?

File: ui/action-menu/src/ActionMenuRoot.tsx [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+  slider?: boolean;
+  density?: DensityProp;
+} & RadixDropdownMenuProps;
+
+export const ActionMenuRoot = ({
+  slider = false,
+  density = "default",
+  ...props
+}: ActionMenuRootProps) => {
+  const [stack, setStack] = React.useState([window?.crypto.randomUUID()]);
```
Comment (wp-aberg) [inline ui/action-menu/src/ActionMenuRoot.tsx:36]: This is part of demos not rendering, even though `window` is conditional Next still doesn't like it and throws an error

File: ui/action-menu/package.json [hunk 9]
```diff
@@ -0,0 +1,54 @@
+{
+  "name": "@washingtonpost/wpds-action-menu",
+  "version": "1.7.0",
```
Comment (wp-aberg) [inline ui/action-menu/package.json:3]: bump to 1.9.1 to be consistent with other packages

File: ui/kit/package.json [hunk 10]
```diff
@@ -32,8 +32,8 @@
     "url": "https://github.com/washingtonpost/wpds-ui-kit/issues"
   },
   "dependencies": {
-    "@washingtonpost/eslint-plugin-wpds": "1.9.1",
     "@washingtonpost/wpds-accordion": "1.9.1",
+    "@washingtonpost/wpds-action-menu": "*",
```
Comment (wp-aberg) [inline ui/kit/package.json:36]: This should also be `1.9.1` as opposed to *

File: build.washingtonpost.com/docs/components/action-menu.mdx [hunk 11]
```diff
@@ -0,0 +1,198 @@
+---
+title: Action Menu
+description: A temporary overlay containing a list of items (commands or actions) relevant to the user’s current context. An item in the menu is immediately executed upon selection. Action menus must be paired with a trigger (button or text link) that reveals the menu on click or tap.
+component: ActionMenu
```
Comment (wp-aberg) [inline build.washingtonpost.com/docs/components/action-menu.mdx:4]: we should include the Draft status here
`status: Draft`

**Review by wp-aberg — APPROVED** [review]
After including the `Draft` status and resolving the merge conflict this should be good to go

**Review by artmsilva — APPROVED** [review]
LGTM! Let's merge this

File: package.json (hunk 1 above)
Comment (ebgranger) [reply package.json]: im not sure on this one, going to add it back and just see what's what

**Pull request merged**